common/commands/config.go

```diff
@@ -143,7 +150,26 @@ func (cc *ConfigCommand) ServerDetails() (*config.ServerDetails, error) {
 }
 
 func (cc *ConfigCommand) CommandName() string {
-	return "config"
+	isOidcConfigured, err := clientUtils.GetBoolEnvValue(coreutils.UsageOidcConfigured, false)
+	if err == nil && isOidcConfigured {
+		return ConfigOidcCommandName
+	}
+	return ConfigCommandName
+}
+
+// ExecAndReportUsage runs the ConfigCommand and then triggers a usage report if needed,
+// Report usage only if OIDC integration was used
+// Usage must be sent after command execution as we need the server details to be set.
+func (cc *ConfigCommand) ExecAndReportUsage() error {
+	err := cc.Run()
+	if cc.CommandName() == ConfigOidcCommandName {
+		channel := make(chan bool)
+		// Triggers the report usage.
+		go reportUsage(cc, channel)
```

Comment:
In this specific case, there's no need to run the `reportUsage` as a go routine, beucase there's nothing to parallelize here. This function's execution is triggered after the  `c.Run()` command anyway. There;s n need for the channel as well.
We can simply have -
```
if err := cc.Run(); err != nil {
   return err
}
reportUsage(cc, nil)
return nil
```
and then we can change the `reportUsage` function to handle a nil channel.